src/components/navigation/section-document-or-browser.tsx

```diff
@@ -172,9 +178,102 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = observer(function Sect
       tabsExtraClassNames={{"chat-open": isChatOpen}}
       onSelect={handleTabSelect}
       selectedIndex={subTabIndex}
-      renderSubTabPanel={subTab =>
-        renderDocumentView(subTab) || renderDocumentBrowserView(subTab)
-      }
+      renderSubTabPanel={subTab => renderDocumentView(subTab) || renderDocumentBrowserView(subTab)}
     />
   );
 });
+
+interface DocumentBrowserScrollerProps {
+  subTab: ISubTabSpec;
+  tabSpec: NavTabModelType;
+  openDocumentKey: string;
+  onSelectDocument: (document: DocumentModelType) => void;
+}
+const DocumentBrowserScroller =
+    ({subTab, tabSpec, openDocumentKey, onSelectDocument}: DocumentBrowserScrollerProps) => {
+  const [scrollerCollapsed, setScrollerCollapsed] = useState(false);
+  const [panelWidth, setPanelWidth] = useState(0);
+  const [scrollWidth, setScrollWidth] = useState(0);
+  const [scrollLeft, setScrollLeft] = useState(0);
```

Comment:
I mentioned this in my last PR review, but I think you want to avoid using state like this if possible. I'm sure there are many cases where the state will get out of sync with what's actually going on, but I was able to find one such case: if you resize the window. Try expanding the left panel to be full screen, or shrinking your window to be very small. You'll find that your state here doesn't stay up to date, and so the scrolling behavior stops working properly.

I tried fixing this by defining a `collectionRef` here which gets passed to `DocumentCollectionList`, then you can define `const scrollWidth = collectionRef.current?.scrollWidth ?? 0;` and similarly for `scrollLeft` and `panelWidth`. This worked pretty well, but not perfectly. I think you'd need to include some additional checks that rerenders the component when the size changes, and maybe when the scroll changes (if that's allowed).